crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -1030,6 +1039,27 @@ fn get_inventory_fragments(
     }
 }
 
+async fn create_tedge_agent_supported_ops(ops_dir: PathBuf) -> Result<(), ConversionError> {
+    create_file_with_defaults(ops_dir.join("c8y_SoftwareUpdate"), None)?;
+    create_file_with_defaults(ops_dir.join("c8y_Restart"), None)?;
+
+    Ok(())
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct HealthStatus {
+    pub pid: Option<u64>,
+    pub status: Option<String>,
```

Comment:
Why `Option`? A health status should definitely have both these fields, right?

Reply:
These topics are open, anything can be published. So, to be safer, if there is no status in the health message then it will not fail.

Reply:
Yeah there is the possibility of anyone publishing anything on that topic. But, as the mapper, it should only parse the well-formed health status messages with a valid `pid` and `status` fields. Any other malformed messages should just be ignored, without crashing the mapper.

Reply:
Yes, right. I did make it String.